### PR TITLE
Add DeviceInputSystem to Input Manager

### PR DESCRIPTION
**Pull request by PolygonalSun**
This PR contains all of the changes to update the InputManager (and other systems using the InputManager) to use the DeviceInputSystem for its event foundation.  This is a remake of a previous PR with changes to account for the jump to 5.0 (https://github.com/BabylonJS/Babylon.js/pull/9194)

### CODE REVIEW

File: src/Cameras/Inputs/BaseCameraMouseWheelInput.ts [hunk 1]
```diff
@@ -61,9 +62,9 @@ export abstract class BaseCameraMouseWheelInput implements ICameraInput<Camera>
                 return;
             }
 
-            const event = <MouseWheelEvent>pointer.event;
+            const event = <IWheelEvent>pointer.event;
 
-            const platformScale = event.deltaMode === WheelEvent.DOM_DELTA_LINE ? this._ffMultiplier : 1;
+            const platformScale = event.deltaMode === 0x01 ? this._ffMultiplier : 1;  // If this happens to be set to DOM_DELTA_LINE, adjust accordingly
```
Comment (deltakosh) [inline src/Cameras/Inputs/BaseCameraMouseWheelInput.ts:67]: we need to NOT use unnamed constants

Comment (PolygonalSun) [reply src/Cameras/Inputs/BaseCameraMouseWheelInput.ts]: I changed this to a constant that's maintained in the file that has our event interfaces.

**Review by RaananW — COMMENTED** [review]
LGTM!

The only thing I would ask it to change all `==` to `===` and `!=` to `!==` to keep type safety in equations. Perf will be slightly better as well (though probably negligible)

**Comment by PolygonalSun** [comment]
> LGTM!
> 
> The only thing I would ask it to change all `==` to `===` and `!=` to `!==` to keep type safety in equations. Perf will be slightly better as well (though probably negligible)

Okay, I changed over what I saw in the InputManager and DeviceInputSystem

**Comment by PolygonalSun** [comment]
Tagging @deltakosh and @CedricGuillemet

**Review by deltakosh — APPROVED** [review]

**Pull request merged**